internal/ingest/goa_batch.go

```diff
@@ -13,7 +18,69 @@ func (svc *ingestImpl) AddBatch(
 	ctx context.Context,
 	payload *goaingest.AddBatchPayload,
 ) (*goaingest.AddBatchResult, error) {
-	return nil, ErrNotImplemented
+	if payload == nil {
+		return nil, goaingest.MakeNotValid(errors.New("missing payload"))
+	}
+
+	sourceID, err := uuid.Parse(payload.SourceID)
+	if err != nil {
+		return nil, goaingest.MakeNotValid(errors.New("invalid SourceID"))
+	}
+
+	if len(payload.Keys) == 0 {
+		return nil, goaingest.MakeNotValid(errors.New("empty Keys"))
+	}
+
+	claims, err := checkClaims(ctx)
+	if err != nil {
+		return nil, goaingest.MakeNotValid(err)
+	}
+
+	// TODO: Discuss identifier generation strategy.
+	bUUID := uuid.Must(uuid.NewRandomFromReader(svc.rander))
+	identifier := fmt.Sprintf("Batch-%s", bUUID.String())
+	if payload.Identifier != nil && *payload.Identifier != "" {
+		identifier = *payload.Identifier
+	}
+	b := &datatypes.Batch{
+		UUID:       bUUID,
+		Status:     enums.BatchStatusQueued,
+		Identifier: identifier,
+		SIPSCount:  len(payload.Keys),
+	}
+
+	if claims != nil {
+		b.Uploader = &datatypes.User{
+			UUID:    uuid.Must(uuid.NewRandomFromReader(svc.rander)),
```

Comment:
It seems strange to me to generate a UUID for the user that may or may not be used, I assumed this was going to create a new user every time. 😕  I'm assuming the `perSvc..CreateBatch()` method isn't generating a UUID when needed because we don't want to inject a random number generator, but I think we should review this decision in respect to the making the API more intuitive.